File: tag/Array/283. Move Zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        if(nums.size() < 2) return;
        int fir = 0;
        while(nums[fir] != 0 && fir < nums.size()) fir++;
        int sec = fir;
        while(fir < nums.size()){
            if(nums[fir] != 0) fir++;
            else{
                while(nums[sec] == 0){
                    sec++;
                    // the nail is full of zero
                    if(sec == nums.size()) return;
                }
                swap(nums[fir], nums[sec]);
                fir++;
            }
        }
    }
};

File: java/Array/283. Move Zeroes.java
// Array
public class Solution {
    public void moveZeroes(int[] nums) {
        int count = 0, first = 0, second = 0;
        for(; second < nums.length; second++) {
            if(nums[second] != 0) {
                nums[first] = nums[second];
                first++;
            }
            else{
                count++;
            }
        }
        int len = nums.length;
        for(int i = 0; i < count; i++) {
            nums[len - 1 - i] = 0;
        }
    }
}
